Python/pywarpx/diagnostics/data/BaseReader.py

```diff
@@ -0,0 +1,185 @@
+import collections
+import os
+import re
+
+import numpy as np
+import pandas as pd
+
+
+class DataReader():
```

Comment:
Maybe more descriptive:
```suggestion
class ReducedDiagnosticsData():
```